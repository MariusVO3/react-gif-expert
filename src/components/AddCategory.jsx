import { useState } from 'react';
import PropTypes from "prop-types";

// Componente funcional que recibe una función como prop: onNewCategory
export const AddCategory = ({ onNewCategory }) => {

    // Creamos un estado local para controlar el valor del input
    const [inputValue, setInputValue] = useState('');

    // Manejador del evento onChange del input
    // Actualiza el estado con el valor que el usuario escribe
    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    // Manejador del evento onSubmit del formulario
    // Evita que se recargue la página y ejecuta la lógica de validación/envío
    const onSubmit = (event) => {
        // Previene el comportamiento por defecto del formulario (para que no refresque)
        event.preventDefault(); 

        // Si el texto tiene 1 carácter o menos, no hacemos nada
        if (inputValue.trim().length <= 1) return;

        // Ejecutamos la función pasada por props con el valor del input limpio
        onNewCategory(inputValue.trim());

        // Limpiamos el campo de texto tras enviar
        setInputValue('');
    }

    // Renderizamos el formulario con un input controlado
    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input 
                type="text" // Campo de texto
                placeholder="Buscar gifs" // Texto sugerido dentro del input
                value={inputValue} // Valor actual del input (estado controlado)
                onChange={onInputChange} // Manejador que actualiza el estado al escribir
            />
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}