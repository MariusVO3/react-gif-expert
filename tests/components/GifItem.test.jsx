import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas para GifItem', () => { 
    
    const title = 'Messi';
    const url = 'https://static.messi.com/wp-content/uploads/2019/10/messi-logo-01.png'

    test('debe hacer match con el snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('Debe existir el titulo y url en los props', () => { 
        render(<GifItem title={title} url={url}/>);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    });
 })