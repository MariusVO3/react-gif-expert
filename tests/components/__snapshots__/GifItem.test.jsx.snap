// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas para GifItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Messi"
      src="https://static.messi.com/wp-content/uploads/2019/10/messi-logo-01.png"
    />
    <p>
      Messi
    </p>
  </div>
</div>
`;
